File: app/src/test/java/com/tasktree/BallTest.java
package com.tasktree;

import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class BallTest {

    @Test
    public void givenTwoBallsThatAreEqualThenEqualsReturnsTrue( ) {

        Ball ballOne = new Ball( 1, new Position( 1 ) );
        Ball ballTwo = new Ball( 1, new Position( 1 ) );

        assertTrue( ballOne.equals( ballTwo ) );
        assertTrue( ballTwo.equals( ballOne ) );
    }

    @Test
    public void givenTwoUnequalBallsThenEqualsReturnsFalse( ) {
        Ball ballOne = new Ball( 1, new Position( 0 ) );
        Ball ballTwo = new Ball( 2, new Position( 0 ) );

        assertFalse( ballOne.equals( ballTwo ) );
        assertFalse( ballTwo.equals( ballOne ) );
    }

}

File: app/src/test/kotlin/com/tasktree/BallTest.kt
package com.tasktree

import org.junit.Assert.*
import org.junit.Test

class BallTest {

    @Test
    fun givenTwoBallsThatAreEqualThenEqualsReturnsTrue() {

        val ballOne = Ball(1, Position(1))
        val ballTwo = Ball(1, Position(1))

        assertTrue(ballOne == ballTwo)
        assertTrue(ballTwo == ballOne)
    }

    @Test
    fun givenTwoUnequalBallsThenEqualsReturnsFalse() {
        val ballOne = Ball(1, Position(0))
        val ballTwo = Ball(2, Position(0))

        assertFalse(ballOne == ballTwo)
        assertFalse(ballTwo == ballOne)
    }

    @Test
    fun givenNewBallPositionThenMoveWillMoveBallToNewPosition() {

        val ball = Ball(1, Position(9))
        val newPosition = Position(1)

        assertEquals( Ball(1, newPosition) , ball.moveTo(newPosition))
    }

}